composer/trainer/trainer.py

```diff
@@ -2803,11 +2909,48 @@ def _train_batch(self, use_grad_scaling: bool) -> dict[str, torch.Tensor]:
                     dist.all_reduce(all_ranks_finished_tensor, reduce_operation='MIN')
                     all_ranks_finished = all_ranks_finished_tensor.item() == 1
                 if found_cuda_oom == 1:
+                    # Readd sync hooks if they were previously turned off
+                    if self.state.fsdp_enabled and len(self.automicrobatch_fsdp_hook_handles) == 0:
+                        self.automicrobatch_fsdp_hook_handles = _readd_fsdp_sync_hooks(self.fsdp_modules, sync_hook)
                     _adjust_device_train_microbatch_size(self.state)
+                    self.num_consecutive_thrashes = 0
+                    self.num_consecutive_non_OOM_batches = 0
+                    oom_found_this_batch = True
                     # Skip return and rerun after handling oom
                     continue
+                if not oom_found_this_batch and torch.cuda.is_available():
+                    # Sync across all ranks to check if any rank had additional alloc retries this batch
+                    self.num_consecutive_thrashes = _update_num_consecutive_thrashes(
+                        self.state,
+                        self.num_consecutive_thrashes,
+                        self.cumulative_alloc_retries,
+                    )
+                if self.num_consecutive_thrashes >= 2:
```

Comment:
@mvpatel2000 @JackZ-db A small number of alloc retries consistently increasing, or a large number of alloc retries in the first couple batches might be ok right and we wouldn't want to decrease dtms?

Reply:
correct me if I'm wrong, but allocations should stabilize after 2 batches? I agree that it makes sense to make this threshold more relaxed though - I'm thinking we can increase the number of consecutive thrashes needed or add an additional check saying alloc retries can't go up by `x` amt? @mvpatel2000

Reply:
I've only seen consistently increasing + safe behavior with MoEs with dropless routing. Given auto does not work for MoEs as is, I think it's fine. 

I view this as a more conservative approach (decreasing DTMS too much => 10% slowdown, thrashing => 2x slowdown), and I think that's the right place to start -- if we discover later it's too cautious, we can always tune it.

Reply:
I believe we have also seen this for huge dense models.

Reply:
Have an example you can dig up?

Reply:
@snarayan21 can you share on slack?